libSmeagol/pocket.py

```diff
@@ -255,14 +252,15 @@ def __setAsType(self, key: str, value: Any, to_type: Type) -> None:
     #  @return Returns the value as the new type or the default value in case of failure
     def __castSafe(self, value: Any, to_type: Type, default: Any = None) -> Any:
         try:
-            if (type(bool()) == to_type) and (type(str()) == type(value)):
+            if (to_type is bool) and (type(value) is str):
```

Comment:
This is proper typechecking in python
```suggestion
            if isinstance(to_type, bool) and isinstance(value, str):
```

Reply:
Of course! This was a brainfart  from my side --- I'd have written it your way, but somehow missed it while focusing on simplifying what was already there. Thanks for the correction!

Reply:
Update: actually changed to:
```suggestion
            if to_type is bool and isinstance(value, str):
```